src/UI/Internal/DateInput.elm

```diff
@@ -162,16 +164,19 @@ ifValid if_ then_ value =
             then_
 
 
-toTextField : String -> (String -> msg) -> String -> DateInput -> TextField msg
-toTextField separator editMsg label current =
+toTextField : RenderConfig -> String -> (String -> msg) -> String -> DateInput -> TextField msg
+toTextField cfg separator editMsg label current =
     let
         correct =
             current
                 |> toDD_MM_YYYY separator
                 |> TextField.singlelineText editMsg label
 
+        invalidTerm =
+            cfg |> localeTerms >> .dateInput >> .invalid
```

Comment:
To be honest, not sure if this is better, still quite long, but I'm okay with that anyway hahaha :D